src/main/kotlin/no/nav/syfo/aktivitetskrav/AktivitetskravService.kt

```diff
@@ -79,10 +84,25 @@ class AktivitetskravService(
 
         val updatedAktivitetskrav = aktivitetskrav.vurder(aktivitetskravVurdering = aktivitetskravVurdering)
 
-        aktivitetskravRepository.createAktivitetskravVurdering(
-            aktivitetskrav = updatedAktivitetskrav,
-            aktivitetskravVurdering = aktivitetskravVurdering,
-        )
+        val varselType = aktivitetskravVurdering.getVarselType()
+        if (varselType != null) {
+            val varsel = AktivitetskravVarsel.create(type = varselType, document = document)
+            val pdf = varselPdfService.createVarselPdf(
+                personIdent = aktivitetskrav.personIdent,
+                varsel = varsel,
+                callId = callId,
+            )
+            aktivitetskravVarselRepository.create(
+                aktivitetskrav = updatedAktivitetskrav,
+                varsel = varsel,
+                pdf = pdf,
+            )
```

Comment:
Man kunne kanskje tro at det burde være en vurdering som skulle opprettes eller oppdateres her?

Reply:
Refaktorert slik at ny vurdering sendes inn her i stedet for at den hentes ut igjen fra aktivitetskravet inni metoden, så blir det kanskje litt tydeligere.